packages/core/src/transport/httpRequest.ts

```diff
@@ -55,11 +55,12 @@ export function createHttpRequest(
 }
 
 function sendBeaconStrategy(endpointBuilder: EndpointBuilder, bytesLimit: number, { data, bytesCount }: Payload) {
-  const url = endpointBuilder.build()
+  const beaconUrl = endpointBuilder.build('beacon')
+  const xhrUrl = endpointBuilder.build('xhr')
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** Maybe collocate it with the `sendXHR(xhrUrl, data)` to avoid building the xhrUrl while it might not be used if sendBeacon works.